features/dump_offline_inventories.lua

```diff
@@ -11,13 +11,30 @@ local Config = require 'config'
 local set_timeout_in_ticks = Task.set_timeout_in_ticks
 local config = Config.dump_offline_inventories
 
+local ignored_items_set = {}
+for _ , k in pairs(config.ignored_items) do
+    ignored_items_set[k] = true
+end
+
 local offline_player_queue = {}
 
 Global.register({offline_player_queue = offline_player_queue}, function(tbl)
     offline_player_queue = tbl.offline_player_queue
     config = Config.dump_offline_inventories
 end)
 
+local function move_items(source, target)
+    if not source.is_empty() then
```

Comment:
```suggestion
    if source.valid and not source.is_empty() then
```
I think with the way you changed the checks it's technically possible to get here with only one of the inventories being valid.

Reply:
Since we never let go of control between obtaining the reference and using it, it should never become invalid. However we do need a ``nil`` check at that point. Which gives me an idea to rework the whole control flow of how we handle things. So I will come back to it.